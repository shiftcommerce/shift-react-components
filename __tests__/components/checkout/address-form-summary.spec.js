// Libraries
import React from 'react'

// Components
import AddressFormHeader from '../../../src/components/checkout/address-form-header'
import AddressFormSummary from '../../../src/components/checkout/address-form-summary'

test("renders the addresss summary", () => {
  // Arrange
  const headerTitle = 'test'
  const addressLine1 = '112 some street'
  const city = 'Leeds'
  const firstName = 'John'
  const lastName = 'Smith'
  const postcode= 'LS18DS'

  // Act
  const wrapper = shallow(
    <AddressFormSummary 
      headerTitle={headerTitle}
      addressLine1={addressLine1}
      city={city}
      firstName={firstName}
      lastName={lastName}
      onClick={jest.fn()}
      postcode={postcode}
      showEditButton={true}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(addressLine1)
  expect(wrapper).toIncludeText(city)
  expect(wrapper).toIncludeText(firstName)
  expect(wrapper).toIncludeText(lastName)
  expect(wrapper.find(AddressFormHeader).length).toEqual(1)
  expect(wrapper.find('div')).toMatchSelector('.c-address-form__summary')
})
