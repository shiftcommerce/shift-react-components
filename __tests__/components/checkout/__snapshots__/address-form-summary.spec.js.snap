// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the addresss summary 1`] = `
<Fragment>
  <AddressFormHeader
    collapsed={true}
    onClick={[Function]}
    showEditButton={true}
    title="test"
  />
  <div
    className="o-form__wrapper--collapsed c-address-form__summary"
  >
    <p
      className="u-bold"
    >
      John
       
      Smith
       
    </p>
    <span>
      112 some street
      , 
      Leeds
      , 
      LS18DS
    </span>
  </div>
</Fragment>
`;
